src/recycling.cxx

```diff
@@ -353,13 +360,13 @@ void Recycling::transform(Options& state) {
               // These are NOT communicated back into state and will exist only in this component
               // This will prevent neutrals leaking through cross-field transport from neutral_mixed or other components
               // While enabling us to still calculate radial wall fluxes separately here
-              BoutReal nnguard = SQ(Nn[i]) / Nn[is];
-              BoutReal pnguard = SQ(Pn[i]) / Pn[is];
-              BoutReal tnguard = SQ(Tn[i]) / Tn[is];
+              BoutReal nnguard = SQ(Nnlim[i]) / Nnlim[is];
```

Comment:
Question here: If using `SQ(Nnlim)` / Nnlim then the guard value will never fall below the floor. If `Nn` is lower than `Nnlim` then nnguard will be greater than Nn. A possibility is to use `SQ(Nn[i]) / Nnlim[is]` so that the guard cell value goes to zero as `Nn` goes to zero.